packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart

```diff
@@ -46,434 +44,451 @@ void main() {
+      FakeCommand(command: <String>[
+        artifacts.getArtifactPath(Artifact.engineDartBinary),
+        artifacts.getArtifactPath(Artifact.frontendServerSnapshotForEngineDartSdk),
+        '--sdk-root',
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath) + '/',
+        '--target=flutter',
+        '-Ddart.developer.causal_async_stacks=false',
+        ...buildModeOptions(BuildMode.profile),
+        '--aot',
+        '--tfa',
+        '--packages',
+        '/.packages',
+        '--output-dill',
+        '$build/app.dill',
+        '--depfile',
+        '$build/kernel_snapshot.d',
+        '/lib/main.dart',
+      ], exitCode: 1),
+    ]);
+
+    expect(const KernelSnapshot().build(androidEnvironment), throwsA(isA<Exception>()));
   }));
 
-  test('kernel_snapshot does not use track widget creation on profile builds', () => testbed.run(() async {
-    final MockKernelCompiler mockKernelCompiler = MockKernelCompiler();
-    when(kernelCompilerFactory.create(any)).thenAnswer((Invocation _) async {
-      return mockKernelCompiler;
-    });
-    when(mockKernelCompiler.compile(
-      sdkRoot: anyNamed('sdkRoot'),
-      aot: anyNamed('aot'),
-      buildMode: anyNamed('buildMode'),
-      trackWidgetCreation: false,
-      targetModel: anyNamed('targetModel'),
-      outputFilePath: anyNamed('outputFilePath'),
-      depFilePath: anyNamed('depFilePath'),
-      packagesPath: anyNamed('packagesPath'),
-      mainPath: anyNamed('mainPath'),
-      extraFrontEndOptions: anyNamed('extraFrontEndOptions'),
-      fileSystemRoots: anyNamed('fileSystemRoots'),
-      fileSystemScheme: anyNamed('fileSystemScheme'),
-      linkPlatformKernelIn: anyNamed('linkPlatformKernelIn'),
-      dartDefines: anyNamed('dartDefines'),
-    )).thenAnswer((Invocation _) async {
-      return const CompilerOutput('example', 0, <Uri>[]);
-    });
+  test('KernelSnapshot does not use track widget creation on profile builds', () => testbed.run(() async {
+    globals.fs.file('.packages').writeAsStringSync('\n');
+    final String build = androidEnvironment.buildDir.path;
+    processManager = FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(command: <String>[
+        artifacts.getArtifactPath(Artifact.engineDartBinary),
+        artifacts.getArtifactPath(Artifact.frontendServerSnapshotForEngineDartSdk),
+        '--sdk-root',
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath) + '/',
+        '--target=flutter',
+        '-Ddart.developer.causal_async_stacks=false',
+        ...buildModeOptions(BuildMode.profile),
+        '--aot',
+        '--tfa',
+        '--packages',
+        '/.packages',
+        '--output-dill',
+        '$build/app.dill',
+        '--depfile',
+        '$build/kernel_snapshot.d',
+        '/lib/main.dart',
+      ], stdout: 'result $kBoundaryKey\n$kBoundaryKey\n$kBoundaryKey $build/app.dill 0\n'),
+    ]);
 
     await const KernelSnapshot().build(androidEnvironment);
```

Comment:
These tests are a little bit weird because they seem to be only verifying that `KernelSnapshot().build()` doesn't crash in various circumstances. Is there some other verification that can be done after the `build()` call to check that something went right, and not just that nothing went wrong?

Reply:
These are really testing that build configuration is mapping to command line arguments correctly.  I could make them a bit more "real" by giving FakeCommand some how to execute a callback. that callback could create some files on the memfs, then I can do more assertions around features like the depfile and stuff.

I'm not sure if that is repeating some of the mistakes with the process manager mocks

Reply:
FYI @zanderso